sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -340,13 +339,8 @@ public void populateDisplayData(DisplayData.Builder builder) {
      * Like {@link #to(String)}.
      */
     public Write<T> to(ValueProvider<String> outputPrefix) {
-      return toResource(NestedValueProvider.of(outputPrefix,
-          new SerializableFunction<String, ResourceId>() {
-            @Override
-            public ResourceId apply(String input) {
-              return FileBasedSink.convertToFileResourceIfPossible(input);
-            }
-          }));
+      return toResource(
+          NestedValueProvider.of(outputPrefix, FileBasedSink.convertToFileResourceFunction()));
```

Comment:
TBH I don't think classes should own wrapping up their methods as functions. Since this is where we need the class, let's define it here (fine/good to make it state to not capture everything in the closure)